src/releng_channel_status/releng_channel_status/view.py

```diff
@@ -0,0 +1,50 @@
+from backend_common.cache import cache
+from flask import abort, current_app, request
+from flask.views import MethodView
+from .util import HttpRequestHelper
+
+
+class BaseView(MethodView):
+    def get_user_agent_platform(self):
+        return request.user_agent.platform
+
+    def get_user_agent_locale(self):
+        return request.user_agent.language
+
+
+class ChannelStatusView(BaseView):
+    def __init__(self, *args, **kwargs):
+        super(ChannelStatusView, self).__init__(*args, **kwargs)
+        self.http = HttpRequestHelper(current_app.config.get('BALROG_API_URL'))
+        self.single_rule_endpoint = current_app.config.get(
+            'SINGLE_RULE_ENDPOINT')
+        self.rules_endpoint = current_app.config.get('RULES_ENDPOINT')
+        self.release_endpoint = current_app.config.get('RELEASE_ENDPOINT')
+
+    @cache.memoize()
+    def get(self, rule_alias, product, channel):
+        rule = self._get_rule(rule_alias, product, channel)
+        if not rule:
+            abort(404)
+        release = self._get_release(rule['mapping'])
```

Comment:
Good idea on fetching the release blob 👍 

It's actually needed if you want to know what's the build ID of "Firefox-mozilla-central-nightly-latest". I hadn't thought of that in bug 1386307.